schemas/EGA.common-definitions.json

```diff
@@ -1244,6 +1244,59 @@
+
+      "schema_descriptor": {
+        "type": "object",
+        "title": "Schema descriptor",
+        "description": "This node is intended to be used to describe the schemas and standards that a JSON document was based on. For instance, if a sample.json document was created to be validated against EGA.sample.json schema version 1.0.0, such information should be reflected in this block. This way, both a human and a machine can interpret and validate the JSON document efficiently, without the need of guessing versions.",
+        "additionalProperties": false,
+        "required": ["object_type", "described_by_schema_uri", "object_schema_version", "common_schema_version"],
+        "properties": {
+          "object_type": {
+            "type": "string",
+            "title": "Type of the object",
+            "description": "Type of the object (e.g. 'sample') the JSON document describes.",
+            "enum": [ "experiment", "study", "sample", "individual", "submission", "run", "dataset", "analysis", "policy", "DAC", "ArrayExperiment", "ArrayAssay" ]
+          },
+          "described_by_schema_uri": {
+            "type": "string",
+            "title": "URI of the schema",
+            "description": "URI of the schema (e.g. 'https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.sample.json') that describes the JSON document (e.g. 'my_sample.json')",
+            "pattern": "^https://github\\.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA\\..+\\.json$",
+            "examples": ["https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.sample.json"]
+          },
+          "object_schema_version": {
+            "type": "string",
+            "title": "Version of the object's schema",
+            "description": "The version of the object's schema, the one specific for the object the JSON document corresponds to (e.g. 'EGA.sample.json'), not the common definitions schema's version (i.e. 'EGA.common-definitions.json').",
+            "allOf": [
+              {
+                "title": "Check semantic versioning pattern",
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/semantic-versioning-pattern"
+              }
+            ]
+          },
+          "common_schema_version": {
+            "type": "string",
+            "title": "Version of the common definition's schema",
+            "description": "The version of the common definition's schema, the one containing all shared definitions (i.e. 'EGA.common-definitions.json'), not the one specific to the object described by the JSON document (e.g. 'EGA.sample.json').",
+            "allOf": [
+              {
+                "title": "Check semantic versioning pattern",
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/semantic-versioning-pattern"
+              }
+            ]
+          }
```

Comment:
How does versioning with the common definitions works? Every time you update the common definition schema with a change, do you also have to update all the object schemas that reference it to point to the new one? And if so, does that mean you also have to bump the version of the object schemas?

Reply:
> How does versioning with the common definitions works?

In a similar fashion to the way the object's schema versioning. I don't see how one would have to update all object schemas that reference it, since this information, the ``common_schema_version`` would exist **within the JSON documents** and **not the schemas _per se_**. Perhaps I did not entirely understand the question.

So, for example, if I create a ``sample.json`` document today, and the version of the sample schema was ``2.0.1`` and the common definitions was ``1.2.0``, this node would have the following structure:
````JSON
"schema_descriptor": {
        "object_type": "sample",
        "described_by_schema_uri": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.sample.json",
        "object_schema_version": "2.0.1",
        "common_schema_version": "1.2.0"
    },
````
Now, a few weeks later, the common definitions schema is updated by the EGA to version ``2.0.0``, adding new features and what not. The JSON node I just typed above would not change, since the standard used in the moment I created my JSON document was ``1.2.0``. It would change, though, if I, the submitter, now wanted to add new things to the JSON document I created. If I change the reference I validate my document against, and now I used the common definitions version ``2.0.0``, I would need to change that information within the schema descriptor. 

The key is that for us to validate a JSON document we only require **2 pointers: the object's schema** (and its version), and **common definitions schema** (and its version). Combinations of these could coexist based on our criteria of releases, always encouraging to validate documents against the latest version of the schemas.